core/liquidity/v2/sla.go

```diff
@@ -222,3 +232,31 @@ func (e *Engine) calculateHysteresisFeePenalty(currentPenalty num.Decimal, previ
 
 	return num.MaxD(currentPenalty, periodAveragePenalty)
 }
+
+func (e *Engine) LiquidityProviderSLAStats(now time.Time) []*types.LiquidityProviderSLA {
+	stats := make([]*types.LiquidityProviderSLA, 0, len(e.slaPerformance))
+
+	for partyID, commitment := range e.slaPerformance {
```

Comment:
Given `e.slaPerformance` you're going to have to sort the slice afterwards otherwise it might cause the data-node's history segments to diverge.

Reply:
Haha yes! I wanted to do it but forgot 🤭